packages/core/src/server.jsx

```diff
@@ -147,6 +147,29 @@ export default function(defaultStore = appInitialState, headerConfig, reduxMiddl
             preRenderMiddleware(store, newProps)
               .then(() => {
 
+                // Needs may return a special canonRedirect key. If they do so, process a redirect, using the variables provided
+                // in those objects as variables to substitute in the routes.
+                const redirects = Object.values(store.getState().data).filter(d => d.canonRedirect);
```

Comment:
Instead of running `store.getState()` here, can you grab the `initialState` variable that also does that and move it above this code? If anything, we should probably be using that `initialState` code block intertwined here with your new code because it's also used down there to process any error codes in a need.